src/context/auth/AuthProvider.js

```diff
@@ -0,0 +1,100 @@
+import { createContext, useContext, useReducer } from "react";
+import { useEffect, useState } from "react";
+import { v4 as uuid } from "uuid";
+import { useAxios } from "../../services";
+import { authReducer } from "./authReducer";
+import { Toast } from "../../utils";
+const initialUserAuthState = {
+  userAuthState: {
+    isUserLoggedIn: false,
+    encodedToken: "",
+    user: {},
+  },
+};
+const defaultUser = {
+  _id: uuid(),
+  firstName: "Rishikesh",
+  lastName: "Shinde",
+  email: "test@gmail.com",
+  password: "Test@123",
+};
+const AuthContext = createContext(initialUserAuthState);
+
+const useAuth = () => useContext(AuthContext);
+
+const AuthProvider = ({ children }) => {
+  const [userAuthState, userAuthDispatch] = useReducer(
+    authReducer,
+    initialUserAuthState
+  );
+  const [apiData, setApiData] = useState({
+    apiURL: "",
+    method: "",
+    apiPostData: {},
+  });
+
+  const { isLoaderLoading, serverResponse } = useAxios(
+    apiData.apiURL,
+    apiData.method,
```

Comment:
the method will always be "post" in the case of login or signup, so you can directly define it here

Reply:
unless the useEffect in 'useAxios' has method as a dependency as well, in that case you should keep the method in the state

Reply:
I have a dependency there. Thats why I am keeping it that way.